src/test/java/org/jitsi/meet/test/web/WebTestBase.java

```diff
@@ -174,13 +174,24 @@ private void ensureTwoParticipantsInternal(
     {
         ensureOneParticipant(participantOneMeetURL, participantOneOptions);
 
-        Participant participant
+        WebParticipant firstParticipant = getParticipant1();
+
+        WebParticipant participant
             = joinParticipant(1, participantTwoMeetURL, participantTwoOptions);
 
         participant.waitToJoinMUC(10);
 
         participant.waitForIceConnected();
-        participant.waitForSendReceiveData();
+
+        // if the participant is audio only, if audio is muted we will not
+        // receive any data, so skip download check
+        // if the participant is audio only and audio muted there will be no
+        // data sent
+        participant.waitForSendReceiveData(
```

Comment:
in which tests this is a problem ? I'm trying to think of better solution...

To me it seems it would be better if only the problematic test is making such decision, because it knows it's the case.

Wouldn't it be nicer if Safari WebParticipant, which is aware of it's limitations, if it would ignore the requirement to both send and receive at the same time ?

Reply:
there are cases which need to check only sending, and cases where it can check only receiving and cases where it is fine to check both.
We can add parameters to ensureParticipant methods, or we can be passing it through ParticipantOptions? WDYT?

Reply:
Then those special places can create participants manually, and not use the "ensure" method. They will know whether to call waitForSendReceive with true or false. The "ensure" method should be a convenience method which covers only standard case. Getting "firstParticipant" here just feels like a big complication.